src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -721,55 +759,77 @@ impl<'tcx> Tree {
     /// - the absence of Strong Protectors anywhere in the allocation
     pub fn dealloc(
         &mut self,
-        tag: BorTag,
+        tag: Option<BorTag>,
```

Comment:
It's odd that `dealloc` changes its signature like this but `perform_access` does not.